e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,133 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace IIoTPlatform_E2E_Tests.Orchestrated {
+    using IIoTPlatform_E2E_Tests.TestExtensions;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using RestSharp;
+    using System;
+    using System.Collections.Generic;
+    using System.Dynamic;
+    using System.Linq;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Xunit;
+    using Xunit.Abstractions;
+
+    [TestCaseOrderer(TestCaseOrderer.FullName, TestConstants.TestAssemblyName)]
+    [Collection("IIoT Multiple Nodes Test Collection")]
+    [Trait(TestConstants.TraitConstants.PublisherModeTraitName, TestConstants.TraitConstants.PublisherModeOrchestratedTraitValue)]
+    public class T_TwinBrowseOrchestratedTestTheory {
+        private readonly IIoTMultipleNodesTestContext _context;
+
+        public T_TwinBrowseOrchestratedTestTheory(IIoTMultipleNodesTestContext context, ITestOutputHelper output) {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.OutputHelper = output ?? throw new ArgumentNullException(nameof(output));
+        }
+
+        [Fact, PriorityOrder(0)]
+        public void SwitchToOrchestratedMode() {
+            TestHelper.SwitchToOrchestratedModeAsync(_context).GetAwaiter().GetResult();
+        }
+
+        [Fact, PriorityOrder(1)]
+        public void D1_Discover_OPC_UA_Endpoints() {
+            var endpoints = TestHelper.Registry_GetEndpoints(_context).GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.First();
+
+            Assert.NotNull(endpoint.Id);
+            Assert.Equal("ActivatedAndConnected", endpoint.ActivationState);
```

Comment:
As there is a handshake process between OPC Twin and OPC Server, it might take some time for the endpoint activation state to be `ActivatedAndConnected` and endpoint state be `Ready`. You might need to wait for that to happen.